### PR TITLE
Fix proxy-configuration being ignored on docker create

### PR DESCRIPTION
**Pull request by thaJeztah**
relates to: https://github.com/docker/cli/pull/93
fixes https://github.com/docker/for-linux/issues/369

Proxies configured in config.json were only taking effect when using `docker run`, but were being ignored when using `docker create`.

Before this change:

    echo '{"proxies":{"default":{"httpProxy":"httpProxy","httpsProxy":"httpsProxy","noProxy":"noProxy","ftpProxy":"ftpProxy"}}}' > config.json
    docker inspect --format '{{.Config.Env}}' $(docker --config=./ create busybox)
    [PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin]

With this change applied:

    echo '{"proxies":{"default":{"httpProxy":"httpProxy","httpsProxy":"httpsProxy","noProxy":"noProxy","ftpProxy":"ftpProxy"}}}' > config.json
    docker inspect --format '{{.Config.Env}}' $(docker --config=./ create busybox)
    [NO_PROXY=noProxy no_proxy=noProxy FTP_PROXY=ftpProxy ftp_proxy=ftpProxy HTTP_PROXY=httpProxy http_proxy=httpProxy HTTPS_PROXY=httpsProxy https_proxy=httpsProxy PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin]

Reported-by: Silvano Cirujano Cuesta <Silvanoc@users.noreply.github.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

**- Description for the changelog**

```Markdown
- Fix proxy-configuration being ignored on docker create [docker/cli#1573](https://github.com/docker/cli/pull/1573)
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @dave-tucker PTAL 😅

**Comment by thaJeztah** [comment]
ping @silvin-lubecki @vdemeester PTAL

**Comment by thaJeztah** [comment]
ping @silvin-lubecki @vdemeester PTAL 🙏

File: cli/command/container/create.go [hunk 1]
```diff
@@ -60,7 +61,17 @@ func NewCreateCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
-func runCreate(dockerCli command.Cli, flags *pflag.FlagSet, opts *createOptions, copts *containerOptions) error {
+func runCreate(dockerCli command.Cli, flags *pflag.FlagSet, options *createOptions, copts *containerOptions) error {
+	proxyConfig := dockerCli.ConfigFile().ParseProxyConfig(dockerCli.Client().DaemonHost(), copts.env.GetAll())
+	newEnv := []string{}
```
Comment (silvin-lubecki) [inline cli/command/container/create.go]: nit: `var newEnv []string`

**Comment by thaJeztah** [comment]
> Or if not, at least factorize the common code between the two commands?

Ideally, we would be able to share more code; the reason I did not do this (and instead went for some small amount of duplication), is that I wanted to keep the code changes minimal (in case we wanted to backport), and didn't want to put the cli-configuration handling too deep into the stack;

- `runRun` (cli specific code) converts the CLI options into a more generic `containeroptions`
  - calls `runContainer` (apply "run" specific options)
  - calls `createContainer` with the given options and container-configuration
- `runCreate` (cli specific code) converts the CLI options into a more generic `containeroptions`
  - calls `createContainer` with the given options and container-configuration


In the above, possibly `runRun` and `runContainer ` could be (partially) combined, but perhaps best left to a bigger refactor (we should more clearly define the "hand-off" point between CLI and "API/client" code).

**Comment by silvin-lubecki** [comment]
That makes sense, thank you @thaJeztah 👍

**Pull request merged**